app/controllers/users_controller.rb

```diff
@@ -1,8 +1,16 @@
 class UsersController < ApplicationController
+  before_action :logged_in_user, except:[:index, :edit, :update, :destroy]
+  before_action :correct_user, only: %i(edit update)
+  before_action :admin_user, only: :destroy
+
   def new
     @user = User.new
   end
 
+  def index
+    @users = User.all.page(params[:page]).per(10)
+  end
+
   def show
```

Comment:
Nếu em k tìm thấy user thì sẽ thế nào? Ví dụ như params[:id] = 10000 chẳng hạn

Reply:
- nếu params[:id] = 1000 thì khi đó @user = nil.
- Trong view show của user hiện tại em đang cho in ra @user.name, mà @user = nil thì nó sẽ bắn lỗi ko có method name cho class nill.
- Nhưng nếu trong view em ko cho in ra những thứ liên quan đến class user thì em nghĩ nó sẽ ko bị lỗi gì ạ.